### PR TITLE
Support FunctoolsPartialVariable::get_function, fix NamedTupleVariable::as_proxy and handle call_function in get_fake_values_from_nodes

### PR DESCRIPTION
**Pull request by laithsakka**
partially address https://github.com/pytorch/pytorch/issues/118785
This diff fixes three things:
1. add get_function to FunctoolsPartialVariable note that it will be available only if all args constant otherwise, 
it would throw unimplemented in the call to asPythonConstant.  

2. NamedTupleVariable takes args dispatched not as list ex: NamedTuple(a, b, c) vs NamedTuple([a, b, c]),
 hence fix that by specializing asProxy. 

3. A call to create_arg from within create_proxy, changes a python NamedTuple to a function call node without 
associating an example value! Updated get_fake_values_from_nodes to handle such case. 
 
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119731
* __->__ #119435
* #119314

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/lists.py [hunk 1]
```diff
@@ -556,6 +556,10 @@ def python_type(self):
     def as_python_constant(self):
         return self.python_type()(*[x.as_python_constant() for x in self.items])
 
+    def as_proxy(self):
+        assert self.python_type() is not SizeVariable
```
Comment (laithsakka) [inline torch/_dynamo/variables/lists.py:560]: no need for this line here

**Comment by laithsakka** [comment]
remove more expected success cases

**Comment by laithsakka** [comment]
after rebase on main getting
```
2024-02-09T01:58:08.3162244Z FAILED [0.0827s] test_transformers.py::TestAttnBiasCPU::test_is_causal_equals_upper_left_shape0_cpu - AssertionError: Can't construct an AttrSource without a valid base source
```
for some of the tests will have to check those !

**Review by jansel — APPROVED** [review]

**Review by anijain2305 — APPROVED** [review]

**Pull request closed without merging**